ls1intum/Artemis
# `General`: Improve student statistics layout, show learning goals to students again and align colors on course overview

### Checklist
#### General


#### Client

### Motivation and Context


The current layout of the student statistics view is improbable. @krusche pitched me some ideas about an enhanced design.
These ideas are realized in this PR.
Additionally, the learning goals are currently not displayed to the students. This is the case because the activated route is not able to fetch the course id from the parent. This information is needed for the REST request to the server.
### Description

The following changes have been done:

- **General structure**: The different charts are now aligned below each other. Additionally, every section has its own title and is separated by lines
- **Ordering**: Below the doughnut chart, the Performance in Exercises line chart appears. Below that, the horizontal bar charts are placed
- **Color distribution**: The color legend in the doughnut chart is now moved from the horizontal bar charts to the chart itself. The color legend also ensures translation sensitivity.
- **Chart appearance**: The height of a horizontal bar has been halved
- **Font size**: The font size of the statistical values on the right side of the view have been aligned with the color legend of the charts
- **Color alignment**: The doughnut chart on course overview now provide colors aligned with the reworked color distribution of Artemis.
- **Learning goals**: The learning goals appear in the statistics again.

### Steps for Testing

Prerequisites:
- 1 Course with exercises and participation
- 1 Instructor
- 1 Student
- 1 Programming Exercise with Complaints enabled

1. Log in to Artemis as student
2. Inspect the course overview page. Make sure the color of the doughnut charts are muted
3. Navigate to the student statistics tab
4. Check the new layout. Make sure everything is still readable and nothing is misaligned. Make sure no learning goals are visible
5. Log in as instructor
6. Create a learning goal for a exercise attached to a lecture
7. Log in as student again
8. Navigate to the student statistics tab
9. Make sure the learning goal is visible now
10. Also make sure everything works with the new dark mode
11. If you have suggestions, feel free to share them. Then we can consider them for potential changes

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| `course-statistics.component.ts` | 87.09% | 95.09% |
| `course-card.component.ts` | 88.67% | 96.15% |


### Screenshots


The new colors of the doughnut charts on course overview:
![image](https://user-images.githubusercontent.com/80622272/167255560-39e4da1f-bcc5-43f1-9a83-85778f44c941.png)

First part of the new student's statistics structure:
![image](https://user-images.githubusercontent.com/80622272/167255626-a782f6cf-7e9c-4d46-8e9d-5c06c1c9ba41.png)

Second part:
![image](https://user-images.githubusercontent.com/80622272/167255765-a75010bb-fb29-441d-984d-d8869ebf4297.png)
